src/main/java/gregtech/api/recipes/ingredients/IntCircuitIngredient.java

```diff
@@ -32,12 +34,39 @@ public static void setCircuitConfiguration(ItemStack itemStack, int configuratio
     }
 
     public static int getCircuitConfiguration(ItemStack itemStack) {
-        if (!MetaItems.INTEGRATED_CIRCUIT.isItemEqual(itemStack) ||
-                !itemStack.hasTagCompound()) return 0;
+        if (!isIntegratedCircuit(itemStack)) return 0;
         NBTTagCompound tagCompound = itemStack.getTagCompound();
         return tagCompound.getInteger("Configuration");
     }
 
+    public static boolean isIntegratedCircuit(ItemStack itemStack) {
+        boolean isCircuit = MetaItems.INTEGRATED_CIRCUIT.isItemEqual(itemStack);
+        if (isCircuit && !itemStack.hasTagCompound()) {
+            NBTTagCompound compound = new NBTTagCompound();
+            compound.setInteger("Configuration", 0);
+            itemStack.setTagCompound(compound);
+        }
+        return isCircuit;
+    }
+
+    public static void adjustConfiguration(PlayerInventoryHolder holder, int amount) {
+        ItemStack stack = holder.getCurrentItem();
+        int configuration = IntCircuitIngredient.getCircuitConfiguration(stack);
+        configuration += amount;
+        configuration = MathHelper.clamp(configuration, 0, IntCircuitIngredient.CIRCUIT_MAX);
+        IntCircuitIngredient.setCircuitConfiguration(stack, configuration);
+
+        holder.markAsDirty();
+    }
```

Comment:
This could pass `holder.getCurrentItem()` into the other `adjustConfiguration()` method, and afterwards markDirty.